src/App.tsx

```diff
@@ -1,109 +1,75 @@
-import React from 'react';
+import React, { useState } from 'react';
+
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const optionsPerPage = [3, 5, 10, 20];
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(optionsPerPage[1]);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const firstVisibleItemIndex = (currentPage - 1) * perPage;
+  const lastVisibleItemIndex = Math.min(42, currentPage * perPage);
+
+  const visibleItems = items.slice(firstVisibleItemIndex, lastVisibleItemIndex);
+
+  const handlePageChange = (newPage: number) => {
+    const maxPage = Math.ceil(items.length / perPage);
+
+    if (newPage >= 1 && newPage <= maxPage) {
+      setCurrentPage(newPage);
+    }
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(+event.target.value);
+    setCurrentPage(1);
```

Comment:
When you change the number of items per page, it's possible for the current page to no longer exist (for example, if you're on the last page and choose to show more items per page). You should add a check to ensure that the current page is within the valid range after changing the 'perPage' state.